lib/plugins/aws/customResources/index.js

```diff
@@ -187,14 +187,21 @@ function addCustomResourceToService(awsProvider, resourceName, iamRoleStatements
           functionName
         );
         customResourceFunction.DependsOn.push(customResourceLogGroupLogicalId);
-        Object.assign(Resources, {
+        const resourcesLogGroup = {
           [customResourceLogGroupLogicalId]: {
             Type: 'AWS::Logs::LogGroup',
             Properties: {
               LogGroupName: awsProvider.naming.getLogGroupName(absoluteFunctionName),
             },
           },
-        });
+        };
+        const logRetentionInDays = awsProvider.getLogRetentionInDays();
+        if (logRetentionInDays) {
+          resourcesLogGroup[
+            customResourceLogGroupLogicalId
+          ].Properties.RetentionInDays = logRetentionInDays;
+        }
+        Object.assign(Resources, resourcesLogGroup);
```

Comment:
It's fine to leave previous configuration and just add:

```
RetentionInDays: awsProvider.getLogRetentionInDays()
```

Note, that if `provider.logRetentionInDays` was not set, the `undefined` will be assigned and such properties are not serialized to CF template JSON, hence ignored.

Reply:
> `undefined` will be assigned and such properties are not serialized to CF template JSON, hence ignored.

I see. Alright I will refactor it. 🙂